SDWebImageSwiftUI/Classes/ImagePlayer.swift

```diff
@@ -87,12 +87,16 @@ public final class ImagePlayer : ObservableObject {
         }
         if let imagePlayer = SDAnimatedImagePlayer(provider: animatedImage) {
             imagePlayer.animationFrameHandler = { [weak self] (index, frame) in
-                self?.currentFrameIndex = index
-                self?.currentFrame = frame
-            }
-            imagePlayer.animationLoopHandler = { [weak self] (loopCount) in
-                self?.currentLoopCount = loopCount
+                // Make sure we trigger combine in the main thread
+                DispatchQueue.main.async {
+                    self?.currentFrameIndex = index
+                    self?.currentFrame = frame
+                }
```

Comment:
down below line 107, seems like we are assigning the runloop mode.. by default it is .common, may be change it there to .main and you might not have to make a dispatchQueue.main block here?

BTW, why are we putting things in main run loop?

Reply:
If you look at SDWebImageSwiftUI comment, you'll see this line:
```
This is useful for some cases, for example, always specify NSDefaultRunLoopMode, if you want to pause the animation when user scroll (for Mac user, drag the mouse or touchpad)
```

we use default to to pause animation when scrolling.

currentFrame is a published object, looks like some times it would randomly freeze if we don't update in UI thread. After making this change, I don't see the issue any more.